docs/keepers.md

```diff
@@ -0,0 +1,51 @@
+# Keepers
+
+## Overview
+
+## Common Conventions
+
+The Keeper class follows similar conventions as the smart contracts. See [smart_contracts.md](./smart_contracts.md) for explanations.
+
+## Classes
+### Keeper
+
+The base interface for all other keeper classes. `generate_actions_for_timestep` returns an array of `action` objects that is run by [experiment.py](../experiments/experiment.py)
+
+### VaultKeeper
+
+Class representation of a keeper who opens vaults. `open_max_vaults` will open the maximum possible number of vaults for each ilk.
+
+### NaiveVaultKeeper
+
+The Naive Vault Keeper will open the maximum number of vaults possible at each timestep.
+
+### AuctionKeeper
+
+The base interface for auction keeper classes. 
+`find_bids_to_place` will take `now` (current timestep as String TODO) as input and will output a dictionary with `ilk_ids` as keys and lists of bids as values.
+`run_bidding_model` will take `bid`, `ilk_id` as inputs and will output a dictionary with "price" as a key and the bid price as the value.
+`place_bid` will take `bid_id`, `price`,`ilk_id`, `now` as inputs and will output an `action` object.
```

Comment:
it outputs a specific `action` object here, though, a `TEND` or `DENT` (or other bid-placing action). i think this is worth noting. it'd also help to explain that the purpose of this function is to figure out what sort of bid to place (tend vs dent) and whether or not it can be placed (is greater than previous bid, etc.)

Reply:
Added